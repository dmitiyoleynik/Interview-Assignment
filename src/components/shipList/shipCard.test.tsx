import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom';

import ShipCard from 'src/components/shipList/shipCard';

test('ShipCard should match snapshot', () => {
  const props = {
    ship: {
      id: 2,
      built: 2017,
      name: 'OOCL Germany',
      lengthMeters: 399.87,
      beamMeters: 58.8,
      maxTEU: 21413,
      owner: 'OOCL (Hong Kong)',
      grossTonnage: 210890,
    },
  };

  const component = renderer.create(
    <MemoryRouter>
      <ShipCard {...props} />
    </MemoryRouter>,
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
