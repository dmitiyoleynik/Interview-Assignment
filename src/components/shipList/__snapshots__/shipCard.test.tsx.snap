// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShipCard should match snapshot 1`] = `
<a
  className="makeStyles-link-2"
  href="/ships/2"
  onClick={[Function]}
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root makeStyles-card-1 css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      >
        OOCL Germany
      </h6>
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Country: 
        Hong Kong
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        TEU: 
        21413
      </p>
    </div>
  </div>
</a>
`;
